lib/phx_tools_web/plugs/curl_detector.ex

```diff
@@ -0,0 +1,44 @@
+defmodule PhxToolsWeb.CurlDetector do
+  @moduledoc """
+  A plug that serves a script based on the request's `User-Agent` or specific URL paths.
+
+  Requests from `curl` receive the script content directly, halting further processing. Additionally, requests to `/Linux.sh` or `/macOS.sh` paths return the script regardless of `User-Agent`.
+  """
+
+  import Plug.Conn
+
+  @spec init(Keyword.t()) :: nil
+  def init(_opts), do: nil
+
+  @spec call(Plug.Conn.t(), nil) :: Plug.Conn.t()
+  def call(conn, _opts) do
+    user_agent =
+      conn
+      |> get_req_header("user-agent")
+      |> List.first()
+
+    if should_serve_script?(user_agent) do
```

Comment:
Now you can call `request_sent_from_curl?/1` directly (notice the function rename).